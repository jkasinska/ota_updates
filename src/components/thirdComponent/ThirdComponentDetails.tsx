import { NavigationProp, RouteProp } from '@react-navigation/native';
import React from 'react';
import { Text } from 'react-native';
import { ThirdComponentStackParamList } from '../../../App';

interface ThirdComponentDetailsProps {
  navigation: NavigationProp<any>;
  route: RouteProp<ThirdComponentStackParamList, 'ThirdComponentDetails'>;
}

const ThirdComponentDetails = (props: ThirdComponentDetailsProps) => {
  return (
    <>
      <Text>ThirdComponentDetails</Text>
      <Text>{props.route.params.title}</Text>
    </>
  );
};

export default ThirdComponentDetails;
