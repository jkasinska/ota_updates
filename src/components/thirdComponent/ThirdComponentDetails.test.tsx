import { render } from '@testing-library/react-native';
import React from 'react';
import { reactNavigationMock } from '../../../__tests__/mocks/reactNavigation.mock';
import ThirdComponentDetails from './ThirdComponentDetails';

describe('ThirdComponentDetails should', () => {
  it('match the snapshot', () => {
    const component = render(
      <ThirdComponentDetails
        {...reactNavigationMock}
        route={{
          key: '',
          name: 'ThirdComponentDetails',
          params: {
            title: 'testTitle',
            key: '',
            id: '',
          },
        }}
      />,
    );
    expect(component).toMatchSnapshot();
  });
});
