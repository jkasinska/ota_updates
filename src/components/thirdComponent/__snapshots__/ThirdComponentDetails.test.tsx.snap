// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThirdComponentDetails should match the snapshot 1`] = `
Array [
  <Text>
    ThirdComponentDetails
  </Text>,
  <Text>
    testTitle
  </Text>,
]
`;
